drivers/usb/device/usb_dc_mcux_nxp.h

```diff
@@ -0,0 +1,372 @@
+/*
+ * Copyright (c) 2018 NXP Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "usb_spec.h"
+
+/******************************************************************************
+ * Definitions
+ *****************************************************************************/
+/* KHCI instance count */
+#ifdef CONFIG_USB_DC_NXP_KHCI
+    #define USB_DEVICE_CONFIG_KHCI (1U)
+#else
+    #define USB_DEVICE_CONFIG_KHCI (0U)
+#endif
+
+/* EHCI instance count */
+#ifdef CONFIG_USB_DC_NXP_EHCI
+    #define USB_DEVICE_CONFIG_EHCI (1U)
+#else
+    #define USB_DEVICE_CONFIG_EHCI (0U)
+#endif
+
+/* IP3511 FS instance count */
+#ifdef CONFIG_USB_DC_NXP_IP3511FS
+    #define USB_DEVICE_CONFIG_LPCIP3511FS (1U)
+#else
+    #define USB_DEVICE_CONFIG_LPCIP3511FS (0U)
+#endif
+
+/* IP3511 HS instance count */
+#ifdef CONFIG_USB_DC_NXP_IP3511HS
+    #define USB_DEVICE_CONFIG_LPCIP3511HS (1U)
+#else
+    #define USB_DEVICE_CONFIG_LPCIP3511HS (0U)
+#endif
+
+#define SETUP_DATA_STAGE_DONE (0)
+#define SETUP_DATA_STAGE_IN (1)
+#define SETUP_DATA_STAGE_OUT (2)
+
+/* Macro to define controller handle */
+typedef void *usb_device_handle;
+#define usb_device_controller_handle usb_device_handle
+
+/* controller driver do the ZLP for controler transfer automatically or not */
+#define USB_DEVICE_CONTROLLER_AUTO_CONTROL_TRANSFER_ZLP (0)
+
+/* endpoint related macros */
+#define EP0_MAX_PACKET_SIZE    64
+#define EP0_OUT 0
+#define EP0_IN  0x80
+/* USB endpoint mask */
+#define USB_ENDPOINT_NUMBER_MASK (0x0FU)
+/* The setup packet size of USB control transfer. */
+#define USB_SETUP_PACKET_SIZE (8U)
+/* Then endpoint number/index calculation */
+#define EP_ADDR2IDX(ep)     ((ep) & ~USB_EP_DIR_MASK)
+#define EP_ADDR2DIR(ep)     ((ep) & USB_EP_DIR_MASK)
+#define EP_ABS_IDX(ep)   (((ep) & ~USB_EP_DIR_MASK) * 2 + (((ep) & USB_EP_DIR_MASK) >> 7))
+
+/* enter critical macros */
+#define USB_OSA_SR_ALLOC() int usbOsaCurrentSr
+#define USB_OSA_ENTER_CRITICAL() usbOsaCurrentSr = irq_lock()
+#define USB_OSA_EXIT_CRITICAL() irq_unlock(usbOsaCurrentSr)
+
+#if defined(CONFIG_BIG_ENDIAN)
+#define SWAP2BYTE_CONST(n) ((((n) & 0x00FFU) << 8U) | (((n) & 0xFF00U) >> 8U))
+#define SWAP4BYTE_CONST(n) ((((n) & 0x000000FFU) << 24U) | (((n) & 0x0000FF00U) << 8U) | (((n) & 0x00FF0000U) >> 8U) | (((n) & 0xFF000000U) >> 24U))
```

Comment:
Use the macros in `include/misc/byteorder.h` instead

Reply:
@MaureenHelm thanks, I have updated it.